### ISSUE TITLE
Request: disable all plugins, and enabling only specific ones

### ISSUE THREAD

**Issue by waldyrious** [svg/svgo#598]
This could be implemented with an option (one that says that we only want to use the explicitly enabled plugins), but I think it would make more sense to be a preset (#312).

For now, the workaround that I've found is to create an empty config file, and use it in conjunction with `--enable`:

`echo "plugins:" > c && svgo --config=c --enable=<plugin_name> test.svg`

**Comment by kevva** [comment]
I would also like this. Sometimes I don't want *all* the optimizations.

**Comment by GreLI** [comment]
There is config with `full: true` option exists for such a purpose.

**Comment by waldyrious** [comment]
I see, that info can be found in [docs/how-it-works/en.md#1-config](https://github.com/svg/svgo/blob/master/docs/how-it-works/en.md#1-config). However, that still requires the creation of a config file, so it doesn't change the current situation much:

`echo "full: true" > c && svgo --config=c --enable=<plugin_name> test.svg`

It would be nice to be able to do this in a single command.

(by the way, `full: true` is very unintuitive; I'd never guess it did that if I saw a config file with it, and hadn't read the "how it works" document previously)

**Comment by GreLI** [comment]
@waldyrious, you can pass config in a form of JSON in cli:
```bash
svgo --config='{"full":true}' --enable=<plugin_name> test.svg
```
but yeah, I agree it's clumsy. Suggestions are welcome!

**Comment by waldyrious** [comment]
Uh. I was pretty sure I had tried passing a config string, but maybe the syntax was wrong and the error message led me to think that the problem was that it expected a file. I'll do some more experiments and will get back to you.

**Comment by waldyrious** [comment]
I guess I just took the documentation's word and didn't even try a string xD

I would suggest the documentation to be changed from

    --config=CONFIG : Config file to extend or replace default

to

    --config=CONFIG : Configuration file (or string) to extend or replace the default

And maybe renaming the "full:true" feature to something else, since if you hadn't told me, I'd have assumed it did the opposite...

**Comment by alinnert** [comment]
IMO, this should be default behaviour. **a)** Most plugin based tools work like that (Webpack, Babel, PostCSS, eslint, tslint...) **b)** You cannot know what the user really needs or wants to do **c)** It's frustrating to figure out why your SVG file is empty, find that plugins are enabled by default and then try to find those plugins that cause this issue, but you have no idea what exactly is going on in the process. (That's what happened to me. I saw that comment that recommends to disable certain plugins. But it seemed they were not enough. Finally, `full: true` worked for me).

I think, enabling plugins by default is not a good idea at all. Especially destructive ones, and those that destroy SVG sprites like the following ones: https://github.com/FortAwesome/Font-Awesome/tree/master/sprites

**Comment by chrispitude** [comment]
This issue is exactly what I was looking for! I need to run *only* three plugins and disable the rest. This works:

```
svgo 1.svg -o 3.svg --pretty --disable={addAttributesToSVGElement,addClassesToSVGElement,cleanupAttrs,cleanupEnableBackground,cleanupIDs,cleanupListOfValues,cleanupNumericValues,collapseGroups,convertColors,convertEllipseToCircle,convertPathData,convertShapeToPath,convertStyleToAttrs,convertTransform,inlineStyles,mergePaths,minifyStyles,moveElemsAttrsToGroup,moveGroupAttrsToElems,prefixIds,removeAttributesBySelector,removeAttrs,removeComments,removeDesc,removeDimensions,removeDoctype,removeEditorsNSData,removeElementsByAttr,removeEmptyAttrs,removeEmptyContainers,removeEmptyText,removeHiddenElems,removeMetadata,removeNonInheritableGroupAttrs,removeOffCanvasPaths,removeRasterImages,removeScriptElement,removeStyleElement,removeTitle,removeUnknownsAndDefaults,removeUnusedNS,removeUselessDefs,removeUselessStrokeAndFill,removeViewBox,removeXMLNS,removeXMLProcInst,reusePaths,sortAttrs,sortDefsChildren} --enable={collapseGroups,moveGroupAttrsToElems,
cleanupIDs}
```

but this does not:

```
svgo 1.svg -o 3.svg --pretty --config='{"full":true}' --enable={collapseGroups,moveGroupAttrsToElems,cleanupIDs}
```

Did the syntax for this solution change?

**Comment by waldyrious** [comment]
@TrySound I appreciate you tackling this! It wasn't initially clear when you closed the issue that it had been resolved, but from reading the [2.0.0 release notes](https://github.com/svg/svgo/releases/tag/v2.0.0) I realized that it was indeed the case:

> In v1 `full` flag allowed to disable all default plugins and run only specified in plugins list. In v2 it's default behaviour.

Thank you!

By the way, just for future reference, it seems to me that this was implemented in 6cecd76f6. Am I correct?
